internal/querynodev2/segments/segment.go

```diff
@@ -199,6 +199,16 @@ func (s *LocalSegment) isValid() bool {
 	return s.ptr != nil
 }
 
+// Provide ONLY the read lock operations,
+// don't make `mut` public to avoid abusing of the mutex
+func (s *LocalSegment) RLock() {
+	s.mut.RLock()
```

Comment:
after RLock,
need to check if segment.ptr == nil?
what happens if we get the lock but delete Segment alread done?

Reply:
It should fail if got a `nil` pointer, will fix this